### PR TITLE
feat: Markdown renderer

### PR DESCRIPTION
**Pull request by PuruVJ**
Moves the rendering logic from svelte.dev here, to be used by kit.svelte.dev as well

Related PRs: 
https://github.com/sveltejs/svelte/pull/8757
https://github.com/sveltejs/kit/pull/10187

### CODE REVIEW

File: packages/site-kit/src/lib/markdown/index.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+/**
+ * @typedef {{
+	name: string;
+	snippet: string;
+	comment: string;
+	deprecated?: string;
+	bullets?: string[];
+	children?: ModuleChild[];
+}} ModuleChild
+ */
```
Comment (benmccann) [inline packages/site-kit/src/lib/markdown/index.js:34]:
```suggestion
/**
 * @typedef {{
 *   name?: string;
 *   comment?: string;
 *   exempt?: boolean;
 *   types?: ModuleChild[];
 *   exports?: ModuleChild[];
 * }[]} Modules
 */

/**
 * @typedef {{
 *   name: string;
 *   snippet: string;
 *   comment: string;
 *   deprecated?: string;
 *   bullets?: string[];
 *   children?: ModuleChild[];
 * }} ModuleChild
 */
```

File: packages/site-kit/src/lib/markdown/renderer.js [hunk 2]
```diff
@@ -0,0 +1,975 @@
...
+		});
+}
+
+/**
+ * Transforms a JS code block into a TS code block by turning JSDoc into type annotations.
+ * Due to pragmatism only the cases currently used in the docs are implemented.
+ * @param {string} js_code
+ * @param {string} [indent]
+ * @param {string} [offset]
+ *  */
```
Comment (benmccann) [inline packages/site-kit/src/lib/markdown/renderer.js:286]:
```suggestion
 */
```

File: packages/site-kit/src/lib/markdown/renderer.js [hunk 3]
```diff
@@ -0,0 +1,975 @@
...
+							} else {
+								code.appendLeft(variable_statement.name.getEnd(), `: ${name}${generics ?? ''}`);
+							}
+
+							modified = true;
+						} else {
+							throw new Error('Unhandled @type JsDoc->TS conversion: ' + js_code);
+						}
+					} else if (ts.isJSDocParameterTag(tag) && ts.isFunctionDeclaration(node)) {
+						// if (node.parameters.length !== 1) {
```
Comment (benmccann) [inline packages/site-kit/src/lib/markdown/renderer.js:362]: remove this? or add a comment about why it's disabled

File: packages/site-kit/src/lib/markdown/renderer.js [hunk 4]
```diff
@@ -0,0 +1,975 @@
...
+}
+
+/** @param {string} html */
+function replace_blank_lines(html) {
+	// preserve blank lines in output (maybe there's a more correct way to do this?)
+	return html.replaceAll(/<div class='line'>(&nbsp;)?<\/div>/g, '<div class="line">\n</div>');
+}
+
+/**
+ *
```
Comment (benmccann) [inline packages/site-kit/src/lib/markdown/renderer.js:848]:
```suggestion
```

**Review by benmccann — APPROVED** [review]
I think some tests could be helpful at some point. I'm a little worried about modifying it since it affects multiple different sites with slightly different setups. But that can be a post-launch item

**Comment by PuruVJ** [comment]
Ah yeah, tests are reallyyyy needed for this stuff, its too complicated and brittle. That is the cost of serving 2 sites at once though. Post launch, hopefully we can modify this logic to have learn.svelte.dev also be able to use it, with the weird +++ and --- syntax

**Pull request merged**